### DISCUSSION TITLE
Bug Report: Modcharts aren't working.

### DISCUSSION THREAD

**Discussion by ILuvGemz** [KadeArchive/Kade-Engine#390] [Q&A]
### If you are playing a downloaded version of the game, what operating system are you using?
Windows (`x86`), Windows (`x86_64`), Linux, or macOS? Specify below.

## Please describe your issue. Provide extensive detail and images if possible.

I have done exactly what the Modcharts.md told me to do. I had put the modchart into assets\data\(song name), then named it modchart.lua, with spelling double-checked, but I wasn't able to make the mods work. I have tried to find the problem, but it didn't work. I couldn't find the issue here. Is there something I had done wrong, or is an actual issue?

![image](https://user-images.githubusercontent.com/76889022/117529556-5bb5db00-b002-11eb-9ce9-23794e6f01d0.png)

https://user-images.githubusercontent.com/76889022/117529599-915ac400-b002-11eb-83ff-852e942cf8cc.mp4

**Comment by Kade-github** [comment]
can you send the modchart.lua?

**Reply by ILuvGemz** [reply]
Here's what it contains:

local sideFlip = false
local downFlip = false
local sideHit = false
local downHit = false
local itgBeat = false
local itgFloat = false
local itgDrift = false
local noteFlip = false
local noteInvert = false

local reflectVal = 1 
local flipPow = 1
local invertPow = 1


function reflector(i)
	if i < 4 then
		reflectVal = -1
	end
end

function start(song)
end

function update(elapsed)
	-- step == 0
	-- step == 4
	if sideFlip = true then
		if sideHit = false then
			for i=0,3 do
				tweenPosXAngle(i, _G['defaultStrum'..i..'X'] + 550,0,0.2142)
			end
			for i=4,7 do
				tweenPosXAngle(i, _G['defaultStrum'..i..'X'] - 550,0,0.2142)
			end
			sideHit = true
		end
		elseif sideHit = true then
			for i=0,3 do
				tweenPosXAngle(i, _G['defaultStrum'..i..'X'] - 550,0,0.2142)
			end
			for i=4,7 do
				tweenPosXAngle(i, _G['defaultStrum'..i..'X'] + 550,0,0.2142)
			end
			sideHit = false
		end
	end
	if downFlip = true then
		if downHit = false then
			for i=0,7 do
				tweenPosYAngle(i, _G['defaultStrum'..i..'Y'] - 165,0,0.2142)
			end
			downscroll = true
			downHit = true
		end
		elseif downHit = true then
			for i=0,7 do
				tweenPosYAngle(i, _G['defaultStrum'..i..'Y'] + 165,0,0.2142)
			end
			downscroll = false
			downHit = false
		end
	end
	
	if itgBeat = true then
		if curBeat % 2 == 0 then
			for i=0,7 do
				setActorX(getActorX(i) - 20,i)
				tweenPos(i,getActorX(i) + 20,getActorY(i),0.07)
			end
		elseif curBeat % 2 == 1 then
			for i=0,7 do
				setActorX(getActorX(i) + 20,i)
				tweenPos(i,getActorX(i) - 20,getActorY(i),0.07)
			end
		end
	end
	
	if noteFlip = true then
		for i=0,4,4 do
			tweenPosXAngle(i, _G['defaultStrum'..i..'X'] + 252 * flipPow,0,0.15)
		end
		for i=1,5,4 do
			tweenPosXAngle(i, _G['defaultStrum'..i..'X'] + 126 * flipPow,0,0.15)
		end
		for i=2,6,4 do 
			tweenPosXAngle(i, _G['defaultStrum'..i..'X'] - 126 * flipPow,0,0.15)
		end
		for i=3,7,4 do
			tweenPosXAngle(i, _G['defaultStrum'..i..'X'] - 252 * flipPow,0,0.15)
		end
	end
	
	if noteInvert = true then
		for i = 0,6,2 do
			tweenPosXAngle(i, _G['defaultStrum'..i..'X'] + 168 * invertPow,0,0.15)
		end
		for i = 1,7,2 do
			tweenPosXAngle(i, _G['defaultStrum'..i..'X'] - 168 * invertPow,0,0.15)
		end
	end
	
	local currentBeat = (songPos / 1000)*(bpm/60)
	
	if itgFloat = true then
		setActorY(_G['defaultStrum'..0..'Y'] + 32 * math.cos((currentBeat + i*0.25) * (math.pi * (-1)), 0)
		setActorY(_G['defaultStrum'..1..'Y'] - 32 * math.cos((currentBeat + i*0.25) * (math.pi), 1)
		setActorY(_G['defaultStrum'..2..'Y'] - 32 * math.cos((currentBeat + i*0.25) * (math.pi * 3 / 4), 2)
		setActorY(_G['defaultStrum'..3..'Y'] + 32 * math.cos((currentBeat + i*0.25) * (math.pi * (-7) / 8), 3)
	end
end

function beatHit(beat)
	if beat == 0 then
        fourStartX = getActorX(4)
        fiveStartX = getActorX(5)
        sixStartX = getActorX(6)
        sevenStartX = getActorX(7)
    end
	if beat == 0
	or beat == 2
	or beat == 85
	or beat == 87 then
		sideFlip = true
	else
		sideFlip = false
	end
	
	if beat == 93
	or beat == 95 then
		downFlip = true
	else
		downFlip = false
	end
	
	if (beat >= 80 and beat <= 108)
	or (beat >= 112 and beat <= 144) then
		itgBeat = true
	else
		itgBeat = false
	end
end

function stepHit(step)
	
end

**Reply by ILuvGemz** [reply]
sent you the modchart.lua here and on your gmail
